src/paper/paper_upload_tasks.py

```diff
@@ -650,12 +655,28 @@ def celery_create_paper(self, celery_data):
                     "level": paper_concept["level"],
                 },
             )
-            paper.unified_document.hubs.add(concept.hub)
 
-    except Exception as e:
-        print("Failed to save concepts for paper" + str(paper.id))
-        sentry.log_error(e)
-        raise e
+            hubs_to_tag_paper_with.append(concept.hub)
+        except Exception as e:
+            print("Failed to save concepts fo paper" + str(paper.id))
+            print(
+                {
+                    "concept": concept,
+                    "concept.hub": concept.hub,
+                    "paper_concept": paper_concept,
+                }
+            )
+            sentry.log_error(
+                e,
+                message={
+                    "concept": concept,
+                    "concept.hub": concept.hub,
+                    "paper_concept": paper_concept,
+                },
+            )
+
+    if len(hubs_to_tag_paper_with) > 0:
+        paper.unified_document.hubs.add(*hubs_to_tag_paper_with)
```

Comment:
Can we add the hubs to the legacy paper hub too?

Reply:
yeah